analyses/compare-gistic/03-GISTIC-gene-level-tally.Rmd

```diff
@@ -0,0 +1,382 @@
+  #   genes_included: a vector of gene symbols to be used to filter 
+  #                   gene_status_df; intended to be used when method = "exon"
+  # 
+  # Returns:
+  #   gene_tally_df: a data.frame with the following columns
+  #                  1. gene: gene symbol
+  #                  2. gain: number of gains tallied
+  #                  3. loss: number of losses tallied
+  #                  4. neutral: number of copy neutral events tallied
+  #                  5. detection_peak: all peaks a gene symbol mapped to 
+  #                                     via GISTIC, comma-separated
+  
+  # TODO: error-handling throughout this function could probably be a bit more
+  # robust
+  if ((method == "exon") & any(c(is.null(biospecimens_included), 
+                                 is.null(genes_included)))) {
+    stop("The 'exon' method is intended to be used with the 
+         biospecimens_included and genes_included arguments")
+  }
+  
+  # If a vector of biospecimen IDs is provided, filter only to those biospecimen
+  # ids
+  if (!is.null(biospecimens_included)) {
+    gene_status_df <- gene_status_df %>%
+      filter(Kids_First_Biospecimen_ID %in% biospecimens_included)
+  }
+  
+  # If a vector of genes is provided, filter only to those genes
+  if (!is.null(genes_included)) {
+    gene_status_df <- gene_status_df %>%
+      filter(gene %in% genes_included)
+  }
+  
+  gene_tally_df <- gene_status_df %>%
+    # Remove any genes that are "nearest genes." We don't expect them to be *in*
+    # the peak and therefore do not expect them to be comparable to the other
+    # method that we used.
+    filter(str_detect(gene, "\\[", negate = TRUE)) %>%
+    # Count the number of times a copy number alteration occurs in an individual
+    # gene
+    group_by(gene, status) %>%
+    tally() %>%
+    spread(status, n) %>%
+    # Any instances where a copy number alteration is NA are equivalent to a 
+    # count of zero
+    replace_na(list(loss = 0, 
+                    gain = 0))
+  
+  # In the GISTIC case, missing values in the neutral column are the same as
+  # zero counts
+  if (method == "gistic") {
+    gene_tally_df <- gene_tally_df %>%
+      replace_na(list(neutral = 0))
+  }
+  
+  # Because neutral calls are not included in the annotated consensus file,
+  # we can assume the number of neutral calls is the number of samples included 
+  # minus any alterations (gain + loss)
+  if (method == "exon") {
+    gene_tally_df <- gene_tally_df %>%
+       mutate(neutral = length(biospecimens_included) - (gain + loss))
+  }
+  
+  gene_tally_df %>%
+    # Add detection peak information to this data.frame. Peaks can be 
+    # overlapping in GISTIC and contain the same gene symbols, so we need to 
+    # summarize the detection_peak column
+    inner_join(gene_mapping_df %>%
+                 select(gene,
+                        detection_peak),
+               by = "gene") %>%
+    group_by(gene, gain, loss, neutral) %>%
```

Comment:
Still unclear what is happening to `amplification`? Are we intentionally ignoring it?

Reply:
Good point! I think in development, because we are filtering the `focal-cn-file-preparation` files to genes in the GISTIC files, I didn't see any instances of `amplification` but I believe the right way to deal with this is to recode all amplifications as gains in the `focal-cn-file-preparation` specifically when we are tallying things for comparison.

Reply:
Good catch – it didn't matter for the LGAT data that I used in development, but it _did_ matter for the entire cohort.